layouts/html.hbs

```diff
@@ -123,9 +118,36 @@
       window.iframeLoaded = new Promise(resolve => {
         iframeLoadedResolve = resolve;
       });
-      window.answersJWT = new Promise(resolve => {
-        window.initAnswersJWT = resolve;
+      window.RuntimeConfig = {
+        lastUpdated: -1
+      };
+      const updateRuntimeConfig = config => {
+        window.RuntimeConfig = {
+          ...window.RuntimeConfig,
+          ...config
+        };
+      };
+      document.addEventListener('DOMContentLoaded', function () {
+        {{> script/on-document-load}}
+        {{#if global_config.waitForRuntimeConfig}}
+          const runtimeConfigNotProvidedTimeout = setTimeout(() => {
+            console.warn(
+              'Runtime config has not been received within 5 seconds of page load, and "waitForRuntimeConfig" is set to true.\n' + 
+              'Load the experience by calling initAnswersExperience(config).'
+            );
+          }, 5000);
+          window.initAnswersExperience = config => {
+            updateRuntimeConfig(config);
+            clearTimeout(runtimeConfigNotProvidedTimeout);
+            initAnswers();
+          };
+        {{else}}
+          initAnswers();
+        {{/if}}
       });
+      const answersMetadata = {
+        hasInitialized: false
```

Comment:
would it be out of scope to refactor the window.iframeResizer logic into some static js file? maybe we can also do something about passing RuntimeConfig over the window?

Reply:
Can you give a little more info on how that would work?

Reply:
sorry I didn't mean to make it sound like it was easy, and didn't have really concrete plans in mind. I was imagining moving this DOMContentLoaded listener into a static file (the on-document-loaded partial would probably stay here. we can just make them separate listeners). For the RuntimeConfig, instead of getting it from the window in core.hbs, maybe we could have a HitchhikerJS.getRuntimeConfig() ? I also thought we might want to refactor window.iFrameResizer into it's own file, but looking at it again that can probably wait.

Reply:
I updated the new code to be inside static files rather than in the handlebars. I will leave the refactoring of the window.iFrameResizer to another PR since that is out of scope here. Changing it to HitchhikerJS.getRuntimeConfig() would also change the public interface for accessing the RuntimeConfig. Maybe I should check with product on that? Personally I think that accessing 'window.RuntimeConfig.linkTarget` is a little simpler than 'HitchhikerJS.getRuntimeConfig().linkTarget', since Hitchhikers will be referencing it from inside the cards.